backend/apps/mentorship/management/commands/mentorship_update_comments.py

```diff
@@ -0,0 +1,152 @@
+            module (Module): The module instance to process.
+
+        """
+        gh = get_github_client()
+
+        module_repos = (
+            module.project.repositories.filter(id__isnull=False)
+            .values_list("id", flat=True)
+            .distinct()
+        )
+
+        if not module_repos.exists():
+            self.stdout.write(
+                self.style.WARNING(f"Skipping. Module '{module.name}' has no repositories.")
+            )
+            return
+
+        relevant_issues = Issue.objects.filter(
+            repository_id__in=module_repos, state=Issue.State.OPEN
+        ).distinct()
+
+        self.stdout.write(f"Found {relevant_issues.count()} open issues across repositories")
+
+        for issue in relevant_issues:
+            self.stdout.write(
+                f"Syncing comments for issue #{issue.number} '{truncate(issue.title, 20)}'"
+            )
+            sync_issue_comments(gh, issue)
+            self.process_issue_interests(issue, module)
+
+    def process_issue_interests(self, issue: Issue, module: Module) -> None:
+        """Process interests for a single issue.
+
+        Args:
+            issue (Issue): The issue instance to process.
+            module (Module): The module instance.
+
+        """
+        existing_interests = IssueUserInterest.objects.filter(module=module, issue=issue)
+        existing_user_ids = set(existing_interests.values_list("user_id", flat=True))
+
+        all_comments = issue.comments.select_related("author").order_by(
+            "author_id", "-nest_created_at"
+        )
+
+        interests_to_create = []
+        interests_to_remove = []
+        new_user_logins = []
+        removed_user_logins = []
+
+        user_interest_status: dict[int, dict[str, Any]] = {}
+
+        for comment in all_comments:
+            if not comment.author:
+                continue
+            user_id = comment.author.id
+            entry = user_interest_status.get(user_id)
+            is_match = any(p.search(comment.body or "") for p in INTEREST_PATTERNS)
+            if entry:
+                entry["is_interested"] = entry["is_interested"] or is_match
+            else:
+                user_interest_status[user_id] = {
+                    "is_interested": is_match,
+                    "login": comment.author.login,
+                    "author": comment.author,
+                }
+
+        for user_id, status in user_interest_status.items():
+            is_interested = status["is_interested"]
+            user_login = status["login"]
+            author = status["author"]
+
+            if is_interested and user_id not in existing_user_ids:
+                interests_to_create.append(
+                    IssueUserInterest(module=module, issue=issue, user=author)
+                )
+                new_user_logins.append(user_login)
+            elif not is_interested and user_id in existing_user_ids:
+                interests_to_remove.append(user_id)
```

Comment:
It looks like any comment that doesn't contain `/interested` after the very first one with `/interested` makes a contributor marked as a no longer interested? Am I reading this right?